src/main/java/seedu/duke/Storage.java

```diff
@@ -1,14 +1,196 @@
+                        + " i/" + description;
+                command = getCommand("add", parametersInput);
+
+                ParameterParser.parse(command, parametersInput);
+
+                // amount would be converted into an integer before being used in the addition of transaction locally
+                int amount = Integer.parseInt(splits[2]);
+
+                switch (type) {
+                case "expense":
+                    storedTransactions.addExpenseDuringStorage(description, amount, category, date);
+                    break;
+                case "income":
+                    storedTransactions.addIncomeDuringStorage(description, amount, category, date);
+                    break;
+                default:
+                    throw new StorageInputCorruptedException();
+                }
+            } catch (DateTimeParseException e) {
+                // If the date format is incorrect, which is due to corrupted date information
+                throw new StorageInputCorruptedException();
+            }
+
+        }
+
+    }
+
+
+    /**
+     * Synthesizes the date into dateString , which would be in the correct format to process.
+     *
+     * @param date which would be processed into a string.
+     * @return dateString, in the correct format to be parsed.
+     */
+
+    private String synthesizeDateString(LocalDate date) {
+
+        String dateOfMonth = String.valueOf(date.getDayOfMonth());
+        String month = String.valueOf(date.getMonthValue());
+        String year = String.valueOf(date.getYear());
+        // 10, 100 , 1000 would be renamed to constant once an appropriate name is discussed.
+        if (date.getMonthValue() < 10) {
+            month = "0" + month;
+        }
+        if (date.getDayOfMonth() < 10) {
+            dateOfMonth = "0" + dateOfMonth;
+        }
+        if (date.getYear() < 10) {
+            year = "000" + year;
+        } else if (date.getYear() < 100) {
+            year = "00" + year;
+        } else if (date.getYear() < 1000) {
+            year = "0" + year;
+        }
+
+        String dateString = dateOfMonth + month + year;
+        return dateString;
+    }
+
+    /**
+     * The function called each time there are changes to the transactions Arraylist.
+     *
+     * @param transactions The updated Arraylist.
+     * @throws IOException If there are issues writing to the duke.txt file
+     */
+    public void writeToFile(ArrayList<Transaction> transactions) throws IOException {
+        FileWriter fileWriter = new FileWriter(FILE_PATH);
+        String transactionEntry = "";
+        for (Transaction transaction : transactions) {
+
+            transactionEntry = transaction.getType() + " | " + transaction.getCategory() + " | "
+                    + transaction.getAmount() + " | " + transaction.getDate() + " | "
+                    + transaction.getDescription();
+
+            fileWriter.write(transactionEntry + "\n");
```

Comment:
Avoid using `\n` line breaks in the code as there are compatibility issues between different operating systems. I recommend replacing them with `System.lineSeparator()`.